app/src/test/java/com/edu/uz/currency/currencyapp/retrofit/NbpClientTest.java

```diff
@@ -0,0 +1,67 @@
+package com.edu.uz.currency.currencyapp.retrofit;
+
+import com.edu.uz.currency.currencyapp.retrofit.pojo.SingleCurrency;
+import com.edu.uz.currency.currencyapp.retrofit.pojo.TableCurrency;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.List;
+
+import static org.assertj.core.api.Java6Assertions.*;
+
+public class NbpClientTest {
+
+    private NbpClient nbpClient;
+
+    @Before
+    public void setUp() throws Exception {
+        nbpClient = NbpClient.FactoryNbpClient.getNbpClient();
+    }
+
+    @Test
+    public void getAllCurrencyFromTable() throws Exception {
+        //given
+        final String table = "a";
+        final String expectedTableName = "A";
+
+        //when
+        final List<TableCurrency> tableCurrencies = nbpClient.getAllCurrencyFromTable(table).execute().body();
```

Comment:
Ja bym proponował zrobić może jakąś klasę serwisu, która by opakowywała tego klienta, żeby nie korzystać z niego bezpośrednio. W tym serwisie można by było uprościć API tak, żeby np. nie trzeba było wiedzieć nic o tym z jakiej tabeli to bierzesz, bo w sumie będziemy korzystać tylko z jednej i żeby nie trzeba było takich pociągów robić za każdym razem, tylko żeby to wyglądało np. tak:
`List<Currency> currencies = currencyService.getAllCurrencies();`